relay-profiling/src/lib.rs

```diff
@@ -133,16 +137,20 @@ fn minimal_profile_from_json(data: &[u8]) -> Result<MinimalProfile, ProfileError
     serde_json::from_slice(data).map_err(ProfileError::InvalidJson)
 }
 
-pub fn expand_profile(payload: &[u8]) -> Result<Vec<u8>, ProfileError> {
-    let profile: MinimalProfile = minimal_profile_from_json(payload)?;
-    match profile.version {
+pub fn expand_profile(payload: &[u8]) -> (Option<EventId>, Result<Vec<u8>, ProfileError>) {
```

Comment:
Would a return type like `Result<(EventId, Vec<u8>), ProfileError>` be better here since this leaves the possibility of a return value like `(None, Ok(...))`. It would also clean up the `match` in the processor.

Reply:
It would clean up the `match` in the processor but since I'll have to change the way to handle errors for each `parse_<platform>_profile`, I suspect it won't improve the code overall.

Now, removing the possibility of a `(None, Ok(...))` is interesting, although Rust wouldn't let us compile if we're not taking care of all the cases at which point I believe we'd see the bug. I'll try to rework it but I don't think it'd be a full win.